src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -792,84 +882,23 @@ private void ExecuteBlockActionsCore<TBlockStartAction, TBlockAction, TNodeActio
-
-                    // Catch Exception from the start action.
-                    ExecuteAndCatchIfThrows(
-                        startAction.Analyzer,
-                        data =>
-                        {
-                            data.action(data.context);
-                            data.blockEndActions?.AddAll(data.scope.CodeBlockEndActions);
-                            data.syntaxNodeActions?.AddRange(data.scope.SyntaxNodeActions);
-                        },
-                        (action: codeBlockStartAction.Action, context: blockStartContext, scope: codeBlockScope, blockEndActions: codeBlockEndActions, syntaxNodeActions),
-                        new AnalysisContextInfo(Compilation, declaredSymbol, declaredNode),
-                        cancellationToken);
-                }
-                else
-                {
-                    if (startAction is OperationBlockStartAnalyzerAction operationBlockStartAction)
-                    {
-                        var operationBlockEndActions = blockEndActions as PooledHashSet<OperationBlockAnalyzerAction>;
-                        var operationBlockScope = new HostOperationBlockStartAnalysisScope(startAction.Analyzer);
-                        var operationStartContext = new AnalyzerOperationBlockStartAnalysisContext(
-                            operationBlockScope, operationBlocks, declaredSymbol, semanticModel.Compilation, AnalyzerOptions,
-                            GetControlFlowGraph, declaredNode.SyntaxTree, filterSpan, isGeneratedCode, cancellationToken);
-
-                        // Catch Exception from the start action.
-                        ExecuteAndCatchIfThrows(
-                            startAction.Analyzer,
-                            data =>
-                            {
-                                data.action(data.context);
-                                data.blockEndActions?.AddAll(data.scope.OperationBlockEndActions);
-                                data.operationActions?.AddRange(data.scope.OperationActions);
-                            },
-                            (action: operationBlockStartAction.Action, context: operationStartContext, scope: operationBlockScope, blockEndActions: operationBlockEndActions, operationActions: operationActions),
-                            new AnalysisContextInfo(Compilation, declaredSymbol),
-                            cancellationToken);
-                    }
-                }
-            }
+                addActions(startAction, blockEndActions, argument, cancellationToken);
 
             using var _ = PooledDelegates.GetPooledFunction((d, ct, arg) => arg.self.IsSupportedDiagnostic(arg.analyzer, d, ct), (self: this, analyzer), out Func<Diagnostic, CancellationToken, bool> isSupportedDiagnostic);
 
             // Execute stateful executable node analyzers, if any.
-            if (executableNodeActions.Any())
```

Comment:
.Any moved into the executeActions logic.  it means not having to pass thsi value in here just for this boolean check.